integration-tests/contract-failing-noop.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { InMemorySigner } from "@taquito/signer";
+import { CONFIGS, defaultSecretKey, isSandbox } from "./config";
+import { OpKind, TezosToolkit } from "@taquito/taquito";
+import { verifySignature } from "@taquito/utils";
+
+CONFIGS().forEach(({ rpc, setup}) => {
+
+  const Tezos = new TezosToolkit(rpc);
+  Tezos.setSignerProvider(new InMemorySigner(defaultSecretKey.secret_key));
+  
+  describe(`Test failing_noop through contract api using: ${rpc}`, () => {  
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    it('Verify that the contract.failingNoop signs a text on the genesis block', async (done) => {
+      const signed = await Tezos.contract.failingNoop({
+        arbitrary: "48656C6C6F20576F726C64", // Hello World
+        basedOnBlock: 0,
+      });
+      if (isSandbox({rpc})) {
+        expect(signed).toEqual({
+          bytes: 'e05b338df1b65370c74ca04e600a526b029abd2827b0c7c71389fb542a8ed9dc110000000b48656c6c6f20576f726c64',
+          signature: 'edsigteDrJhTBCX8cX8swZPEeT7N96wjUxWfFPJSEFL5s5pUT2C3C2FDNF169zYYKktrZHwpHcWan6uyKj89tK43fqY2hbbKoxT',
+          signedContent: {
+            branch: await Tezos.rpc.getBlockHash({block: 'genesis'}),
+            contents: [{
+              kind: OpKind.FAILING_NOOP,
+              arbitrary: '48656C6C6F20576F726C64'
+            }]
+          }
+        });
+      } else {
```

Comment:
to handle assertion based on `isSandbox` is limited to one other testnet. Find it easier to manage if we can separate into test suites named by testnet like this `const nairobinet = protocol == Protocols.PtNairobi ? it : it.skip;`   and add a comment behind branch that genesis block hash differ by by chain. When new testnet introduced that we can save time debugging why this test isn't happy